lib/elixir/lib/kernel.ex

```diff
@@ -2609,6 +2609,16 @@ defmodule Kernel do
         quote do: Module.put_attribute(__MODULE__, unquote(name), unquote(arg),
                                        unquote(stack), unquote(line))
 
+      name == :optional_callbacks ->
```

Comment:
I don't think we need to check for this. We allow multiple declarations and then we just merge them together at the end. I will make sure we have a test.

Reply:
Test added here: b28aca17975f41f5205ee9bc134ecd4b0bc5ed44